lib/matplotlib/axes/_axes.py

```diff
@@ -5182,7 +5182,22 @@ def _pcolorargs(funcname, *args, **kw):
             return X, Y, C
 
         if len(args) == 3:
+            # Check x and y for bad data...
             X, Y, C = [np.asanyarray(a) for a in args]
+            for a in X, Y:
+                a = cbook.safe_masked_invalid(a)
+                if funcname == 'pcolormesh' and np.ma.is_masked(a):
+                    raise TypeError(
+                        'x and y arguments to pcolormesh cannot have '
+                        'non-finite values and or be of type '
+                        'numpy.ma.core.MaskedArray with masked values')
+            if funcname == 'pcolormesh':
+                # if it didn't fail the above it is masked but has no
+                # masked values, so strip the mask for pcolormesh
+                if type(X) is np.ma.core.MaskedArray:
```

Comment:
I think all this can be simplified:
```python
C = np.asanyarray(C)
X, Y = [cbook.safe_masked_invalid(x) for x in (X, Y)]
if funcname == 'pcolormesh':
    if np.ma.is_masked(X) or np.ma.is_masked(Y):
        raise ValueError("...")
    X, Y = X.data, Y.data
```